src/directive/rateLimit.ts

```diff
@@ -9,58 +12,135 @@ import {
 import { GraphQLError } from 'graphql';
 import { singleRedisClient } from '../redis';
 import { Context } from '../Context';
+import { logger } from '../logger';
+import { WATERCOOLER_ID } from '../common';
 
 export class CustomRateLimiterRedis extends RateLimiterRedis {
-  constructor(props) {
+  constructor(props: IRateLimiterRedisOptions) {
     super(props);
   }
 
   // Currently not doing any special actions/overrides
   // This was primarily introduced to make debugging easier by logging the details of rate limited queries/mutations after receiving a request
-  consume(key, pointsToConsume, options) {
+  consume(
+    key: string | number,
+    pointsToConsume?: number,
+    options?: { [key: string]: unknown },
+  ) {
     if (process.env.NODE_ENV === 'development') {
-      console.log(`[CONSUME] ${key} for ${pointsToConsume}`);
+      logger.info(`[CONSUME] ${key} for ${pointsToConsume}`);
```

Comment:
We can just use `debug` here without NODE_ENV check and if anyone wants to see logging they can move to debug logging locally.

Reply:
Changed it to be debug and without env check